service/internal/k8sclientcache/cache.go

```diff
@@ -0,0 +1,143 @@
+package k8sclientcache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/kubeconfig/v4"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	gocache "github.com/patrickmn/go-cache"
+	"github.com/spf13/afero"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	expiration = 10 * time.Minute
+)
+
+type Config struct {
+	// Dependencies.
+	Fs        afero.Fs
```

Comment:
What do we need the filesystem for?

Reply:
Good catch, I coped it from `client` resource but we don't need it as you said.